crates/aptos-crypto/src/noise.rs

```diff
@@ -302,42 +305,34 @@ impl NoiseConfig {
         let k = mix_key(&mut ck, &dh_output)?;
 
         // -> s
-        let aead = Aes256Gcm::new(GenericArray::from_slice(&k));
-
-        let msg_and_ad = Payload {
-            msg: self.public_key.as_slice(),
-            aad: &h,
-        };
-        let nonce = GenericArray::from_slice(&[0u8; AES_NONCE_SIZE]);
-        let encrypted_static = aead
-            .encrypt(nonce, msg_and_ad)
+        let aead = LessSafeKey::new(UnboundKey::new(&aead::AES_256_GCM, &k[..]).unwrap());
```

Comment:
what is the unwrap about? how could this fail? can it triggered by malicious peer? we should use expect(..) explaining why this is safe or returning an error if it's not safe

Reply:
I also wonder what does LessSafe/Unbound mean from the library

Reply:
This will fail if there's a bug in our code and the size of `k` is not the expected key size for `AES_256_GCM`:

Specifically, see:
```rust
impl UnboundKey {
    /// Constructs an `UnboundKey`.
    ///
    /// Fails if `key_bytes.len() != algorithm.key_len()`.
    pub fn new(
        algorithm: &'static Algorithm,
        key_bytes: &[u8],
    ) -> Result<Self, error::Unspecified> {
        let cpu_features = cpu::features();
        Ok(Self {
            inner: (algorithm.init)(key_bytes, cpu_features)?,
            algorithm,
            cpu_features,
        })
    }
```

Should I add an `expect()` explaining the reason for failure?

Reply:
LessSafe/Unbound are types for more direct / ninja uses of the library, which we have to engage in because the Noise code was written in such a way that it exposes keys as byte arrays, rather than native `ring` types.